### PR TITLE
ORCA-350: Log JUnit output along with coverage

### PR DESCRIPTION
**Pull request by secretsayan**

### CODE REVIEW

File: src/Domain/Tool/Phpunit/PhpUnitTask.php [hunk 1]
```diff
@@ -267,6 +267,7 @@ protected function runPhpUnit(): void {
       ];
       if ($this->shouldGenerateCodeCoverage()) {
         $command[] = "--coverage-clover={$this->cloverCoverage}";
+        $command[] = "--log-junit={$this->junitLog}";
       }
```
Comment (mglaman) [inline src/Domain/Tool/Phpunit/PhpUnitTask.php:271]: I wouldn't consider the following comment a block, just bringing it up:

JUnit provides a way to report test results. This may be useful for CI systems which can report these results on jobs not running coverage. However, I think that's an edge case which can be opened later. GHA reads PHPUnit output directly, anyways.

**Review by mglaman — APPROVED** [review]

File: src/Domain/Tool/Phpunit/PhpUnitTask.php (hunk 1 above)
Comment (secretsayan) [reply src/Domain/Tool/Phpunit/PhpUnitTask.php:271]: Can you please clarify what do you mean by "report test results"?

Comment (mglaman) [reply src/Domain/Tool/Phpunit/PhpUnitTask.php:271]: That's the point of a JUnit XML. It's a standard way to report test results and display them beyond reading `stdout`.
